internal/topology/variables/clusterclass_variable_validation.go

```diff
@@ -215,9 +225,128 @@ func validateSchema(schema *apiextensions.JSONSchemaProps, fldPath *field.Path)
 		allErrs = append(allErrs, validateSchema(&p, fldPath.Child("properties").Key(propertyName))...)
 	}
 
+	// If any schema related validation errors have been found at this level or deeper, skip CEL expression validation.
+	// Invalid OpenAPISchemas are not always possible to convert into valid CEL DeclTypes, and can lead to CEL
+	// validation error messages that are not actionable (will go away once the schema errors are resolved) and that
+	// are difficult for CEL expression authors to understand.
+	if len(allErrs) > 0 {
+		return allErrs
+	}
+
+	celContext := apiextensionsvalidation.RootCELContext(schema)
+	allErrs = append(allErrs, validateCELExpressions(schema, fldPath, celContext)...)
+
+	return allErrs
+}
+
+func validateCELExpressions(schema *apiextensions.JSONSchemaProps, fldPath *field.Path, celContext *apiextensionsvalidation.CELSchemaContext) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if schema.AdditionalProperties != nil {
+		allErrs = append(allErrs, validateCELExpressions(schema.AdditionalProperties.Schema, fldPath.Child("additionalProperties"), celContext.ChildAdditionalPropertiesContext(schema.AdditionalProperties.Schema))...)
+	}
+
+	if len(schema.Properties) != 0 {
+		for property, jsonSchema := range schema.Properties {
+			propertySchema := jsonSchema
```

Comment:
```suggestion
		for propertyName, propertySchema := range schema.Properties {
			p := propertySchema
```
nit: maybe let's implement this the same way as above, seems slightly easier to read

Reply:
Done.